compiler/rustc_resolve/src/access_levels.rs

```diff
@@ -130,16 +159,21 @@ impl<'r, 'ast> Visitor<'ast> for AccessLevelsVisitor<'ast, 'r> {
 
             // Foreign modules inherit level from parents.
             ast::ItemKind::ForeignMod(..) => {
-                let parent_level =
-                    self.r.access_levels.get_access_level(self.r.local_parent(def_id));
-                self.set_access_level(item.id, parent_level);
+                let parent_id = self.r.local_parent(def_id);
+                self.update_effective_vis(
+                    def_id,
+                    Visibility::Public,
+                    parent_id,
+                    AccessLevel::Public,
+                    span,
+                );
             }
 
             // Only exported `macro_rules!` items are public, but they always are
             ast::ItemKind::MacroDef(ref macro_def) if macro_def.macro_rules => {
-                if item.attrs.iter().any(|attr| attr.has_name(sym::macro_export)) {
-                    self.set_access_level(item.id, Some(AccessLevel::Public));
-                }
+                let parent_id = self.r.local_parent(def_id);
+                let vis = self.r.visibilities.get(&def_id).unwrap().clone();
```

Comment:
```suggestion
                let vis = self.r.visibilities[&def_id].clone();
```
`get` + `unwrap` is just indexing, here and below.